shared/Microsoft.Extensions.Buffers.Sources/BufferReader.cs

```diff
@@ -90,15 +90,34 @@ private void MoveNext()
             _end = true;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public void Advance(int byteCount)
         {
-            if (byteCount < 0)
+            if (byteCount == 0)
```

Comment:
Do we have a test fo 0?

Reply:
Yes, to avoid exception when Advance is called with `0` and `_end == true` ; its moving the method's final exception test up - but is a non exception case when `byteCount == 0`

Reply:
Mentioning tests, gave me an idea - can move the exception out to the non-inlined also and let it handle the `0` case e.g. keep that function the same and simplify the fast path as ?
```csharp
public void Advance(int byteCount)
{
    if (!_end && byteCount > 0 && (_index + byteCount) < _currentSpan.Length)
    {
        _consumedBytes += byteCount;
        _index += byteCount;
    }
    else
    {
        AdvanceNext(byteCount);
    }
}
```

Reply:
I would never merge you PR quickly again :)